src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs

```diff
@@ -2596,5 +2596,197 @@ public void M(string s)
                 Diagnostic(ErrorCode.WRN_IsPatternAlways, "s is string or null").WithArguments("string").WithLocation(7, 13)
                 );
         }
+
+        [Fact]
+        public void SemanticModelForSwitchExpression()
+        {
+            var source =
+@"public class C
+{
+    void M(int i)
+    {
+        C x0 = i switch // 0
+        {
+            0 => new A(),
+            1 => new B(),
+            _ => throw null,
+        };
+        _ = i switch // 1
+        {
+            0 => new A(),
+            1 => new B(),
+            _ => throw null,
+        };
+        D x2 = i switch // 2
+        {
+            0 => new A(),
+            1 => new B(),
+            _ => throw null,
+        };
+        D x3 = i switch // 3
+        {
+            0 => new E(), // 3.1
+            1 => new F(), // 3.2
+            _ => throw null,
+        };
+        C x4 = i switch // 4
+        {
+            0 => new A(),
+            1 => new B(),
+            2 => new C(),
+            _ => throw null,
+        };
+        D x5 = i switch // 5
+        {
+            0 => new A(),
+            1 => new B(),
+            2 => new C(),
+            _ => throw null,
+        };
+        D x6 = i switch // 6
+        {
+            0 => 1,
+            1 => 2,
+            _ => throw null,
+        };
+        _ = (C)(i switch // 7
+        {
+            0 => new A(),
+            1 => new B(),
+            _ => throw null,
+        });
+        _ = (D)(i switch // 8
+        {
+            0 => new A(),
+            1 => new B(),
+            _ => throw null,
+        });
+        _ = (D)(i switch // 9
+        {
+            0 => new E(), // 9.1
+            1 => new F(), // 9.2
+            _ => throw null,
+        });
+        _ = (C)(i switch // 10
+        {
+            0 => new A(),
+            1 => new B(),
+            2 => new C(),
+            _ => throw null,
+        });
+        _ = (D)(i switch // 11
+        {
+            0 => new A(),
+            1 => new B(),
+            2 => new C(),
+            _ => throw null,
+        });
+        _ = (D)(i switch // 12
+        {
+            0 => 1,
```

Comment:
>1 [](start = 17, length = 1)

What TypeInfo do we get for values returned by the switch expression arms?

Reply:
Great question. When there was a switch expression conversion, the converted type of each expression shows the conversion to the final type of the switch expression. I will exhibit that in the tests.

---
In reply to: [440462853](https://github.com/dotnet/roslyn/pull/45069#discussion_r440462853) [](ancestors = 440462853)